docs/universal-gateway/examples/front-door-pattern.mdx

```diff
@@ -0,0 +1,94 @@
+---
+title: "Implement the 'Front Door' Pattern"
+description: By using Cloud Endpoint and internal Agent Endpoints together, you loosely double your gateway and services for extra flexibility.
+sidebar_label: "The Front Door Pattern"
+---
+
+import ReserveDomain from "./snippets/_reserve-domain.mdx";
+import CloudEndpoint from "./snippets/_cloud-endpoint.mdx";
+import Back from "./snippets/_back-to-examples.mdx";
+
+The "front door" pattern involves using a Cloud Endpoint, the forward-internal Traffic Policy action, and an internal Agent Endpoint to route traffic from the public internet to your service(s).
+
+With this pattern, you can:
+
+- Host any number of services under a single hostname (e.g. `your-company.com`).
+- Decouple gateway configuration from per-service traffic management policies.
+- Prevent any driect exposure of your services or their host systems to the public internet.
+- Route traffic and add authentication to microservices with a single API gateway configuration.
+- Authenticate developer environments or CI preview builds behind a single endpoint.
+
+## 1. Create an endpoint for your service
+
+Start an internal [Agent Endpoint](/docs/universal-gateway/agent-endpoints/), replacing `$PORT` based on where your service listens.
+You can also use one of our [SDKs](/docs/agent/sdks) or the [Kubernetes Operator](/docs/k8s).
+
+```bash
+ngrok http $PORT --url https://service.internal
+```
+
+## 2. Reserve a domain
+
+<ReserveDomain />
+
+## 3. Create a Cloud Endpoint
+
+<CloudEndpoint />
+
+## 4. Add routing to your service with Traffic Policy
+
+Copy and paste the policy below into your new Cloud Endpoint in the ngrok dashboard.
+
+```yaml
+on_http_request:
+  - actions:
+      - type: forward-internal
+        config:
+          url: http://service.internal
+```
+
+### What's happening here?
+
+On every HTTP request, the Traffic Policy engine forwards the traffic to the
+internal agent endpoint you created at `https://service.internal`.
+
+## 5. Add a second service and routing (optional)
```

Comment:
Let's move this and 5a out of the steps and turn step six into step five. Since this is optional, we shouldn't include it as part of the steps because it makes the experience fragmented